diff -Naur zchaff/Makefile zchaff_itc/Makefile
--- zchaff/Makefile	2004-11-15 20:44:14.000000000 +0100
+++ zchaff_itc/Makefile	2009-05-07 11:29:04.000000000 +0200
@@ -1,4 +1,5 @@
 CC = g++ -Wall
+CXX = g++ -Wall
 CFLAGS = -O3
 
 #MFLAGS = 
@@ -29,16 +30,16 @@
 
 
 zchaff:   $(SOLVER_OBJS) libsat.a SAT_C.h
-	  $(CC) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) $(SOLVER_OBJS) libsat.a -o zchaff 
+	  $(CXX) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) $(SOLVER_OBJS) libsat.a -o zchaff 
 
 zverify_df: zverify_df.cpp
-	  $(CC) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) zverify_df.cpp -o zverify_df
+	  $(CXX) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) zverify_df.cpp -o zverify_df
 
 zminimal: zminimal.cpp libsat.a
-	  $(CC) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) zminimal.cpp libsat.a -o zminimal
+	  $(CXX) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) zminimal.cpp libsat.a -o zminimal
 
 cnf_stats: cnf_stats.cpp
-	  $(CC) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) cnf_stats.cpp -o cnf_stats
+	  $(CXX) $(LINKFLAGS) $(CFLAGS) $(MFLAGS) cnf_stats.cpp -o cnf_stats
 
 $(LIB_OBJS): $(HEADERS) Makefile
 
@@ -56,7 +57,7 @@
 	$(RANLIB) libsat.a
 
 .cpp.o:
-	$(CC) $(CFLAGS) $(MFLAGS) -c $< 
+	$(CXX) $(CFLAGS) $(MFLAGS) -c $< 
 
 clean:	
 	rm -f *.o libsat.a zchaff *wrapper.cpp zminimal zverify_df cnf_stats SAT_C.h
diff -Naur zchaff/zchaff_dbase.cpp zchaff_itc/zchaff_dbase.cpp
--- zchaff/zchaff_dbase.cpp	2004-11-15 20:44:07.000000000 +0100
+++ zchaff_itc/zchaff_dbase.cpp	2009-05-07 11:30:06.000000000 +0200
@@ -33,6 +33,7 @@
 // of the possibility of those damages.
 // ********************************************************************/
 
+#include <cstdlib>
 #include <iostream>
 #include <vector>
 #include <set>
diff -Naur zchaff/zchaff_utils.cpp zchaff_itc/zchaff_utils.cpp
--- zchaff/zchaff_utils.cpp	2004-11-15 20:44:07.000000000 +0100
+++ zchaff_itc/zchaff_utils.cpp	2009-05-07 11:29:04.000000000 +0200
@@ -38,7 +38,9 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <sys/time.h>
+#ifndef __MINGW32__
 #include <sys/resource.h>
+#endif
 
 int _global_debug_leveli = 0;
 
@@ -80,10 +82,14 @@
 
 double get_cpu_time(void) {
   double res;
+#ifndef __MINGW32__
   struct rusage usage;
   getrusage(RUSAGE_SELF, &usage);
   res = usage.ru_utime.tv_usec + usage.ru_stime.tv_usec;
   res *= 1e-6;
-  res += usage.ru_utime.tv_sec + usage.ru_stime.tv_sec;
+  res += usage.ru_utime.tv_sec + usage.ru_stime.tv_sec; 
+#else 
+  res = 0;
+#endif
   return res;
 }
